### ISSUE TITLE
How to change the volume streaming options from within a viewer/plugin?

### ISSUE THREAD

**Issue by cschlick** [molstar/molstar#934]
Hi, a question. I've enjoyed exploring molstar recently. One thing I am stuck on is how to control the volume streaming options programmatically. For example, changing contour level. When you move the gui sliders it eventually calls [this](https://github.com/molstar/molstar/blob/fc44e66b26d798d09fa681643e65e3d73241838e/src/mol-plugin-ui/custom/volume.tsx#L125C24-L125C24) function in VolumeStreamingCustomControls. But I can't find a path from a high level viewer/plugin object to the instance of that class, or figure out how to set it up differently. I'd really appreciate any tips on the right way to do this.

**Comment by dsehnal** [comment]
The streaming is initialized using [this action](https://github.com/molstar/molstar/blob/fc44e66b26d798d09fa681643e65e3d73241838e/src/mol-plugin/behavior/dynamic/volume-streaming/transformers.ts#L37). If you apply this action from your code, it should return the reference to the results of the  `CreateVolumeStreamingBehavior` transform. You can then use `plugin.build().to(referenceToTheNode).update(newParams).commit()` to apply the changes in a way similar to what you linked.

You should also be able to use `plugin.state.data.selectQ` to search for cells created by the `CreateVolumeStreamingBehavior` transformer (or search for the `VolumeStreaming` "type") to get the `referenceToTheNode`.

**Comment by cschlick** [comment]
Hi David, thanks for the tips. To initialize I am using `this.plugin.runTask(this.state.applyAction(InitVolumeStreaming, this.mapParams, this.assemblyRef));` which I think I found in one of the examples, but neither the inner nor outer function returns a reference for me. Also I wasn't able to get `plugin.state.data.selectQ` to return anything except an empty array for any string input. 

But you pointed me in the right direction and I was able to use `state.select(StateSelection.Generators.ofTransformer(CreateVolumeStreamingBehavior))[0]` to get the reference. Once I had the reference, changing the params and then running `.commit()` like you showed worked well. Thanks again.
